com.civicrm.hrjobroles/CRM/Hrjobroles/Upgrader.php

```diff
@@ -80,6 +80,66 @@ public function upgrade_1005() {
 
     return TRUE;
   }
+  
+  /**
+   * Deletes cost centre "other" option value or disable it if not in use
+   *
+   * @return bool
+   */
+  public function upgrade_1006() {
+    $jobRoles = civicrm_api3('HrJobRoles', 'get');
+    if ($jobRoles['count'] == 0) {
+      $this->deleteCostCentreOther();
+    }
+    
+    $otherId = $this->retrieveCostCentreOtherId();
+    $inUse = FALSE;
+    $roles = $jobRoles['values'];
+    $pattern = '/\|' . $otherId . '\|/';
+    foreach ($roles as $role) {
+      if (preg_match($pattern, $role['cost_center'])) {
+        $inUse = TRUE;
+        break;
+      }
+    }
+    
+    if (!$inUse) { // disables cost centre other
+      civicrm_api3('OptionValue', 'get', [
+        'option_group_id' => 'cost_centres',
+        'name' => 'Other',
+        'api.OptionValue.create' => ['id' => '$value.id', 'is_active' => 0],
+      ]);
+    }
```

Comment:
Just out of curiosity, did anyone ever questioned why we have this? If it is not being used, why can't we delete it? If we don't delete them, it means new sites will be different than existing ones, which is not a good idea. I'd like to know if there's a reason for that

Reply:
It was actually specified that way as part of the requirement. I will have to discuss with @mi-kon